proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/title/GenericTitlePacket.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2018 Velocity Contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.velocitypowered.proxy.protocol.packet.title;
+
+import com.velocitypowered.api.network.ProtocolVersion;
+import com.velocitypowered.proxy.protocol.MinecraftPacket;
+import com.velocitypowered.proxy.protocol.ProtocolUtils;
+import io.netty.buffer.ByteBuf;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public abstract class GenericTitlePacket implements MinecraftPacket {
+
+  public enum ActionType {
+    SET_TITLE(0),
+    SET_SUBTITLE(1),
+    SET_ACTION_BAR(2),
+    SET_TIMES(3),
+    HIDE(4),
+    RESET(5);
+
+    private final int action;
+
+    ActionType(int action) {
+      this.action = action;
+    }
+
+    public int getAction(ProtocolVersion version) {
+      return version.compareTo(ProtocolVersion.MINECRAFT_1_11) < 0
+              ? action > 2 ? action - 1 : action : action;
```

Comment:
This logic is prone to bugs, I would store two different fields for pre-1.11 and post-1.11.

Reply:
This is a clientbount packet and we don't expect the server to send wrong data. I'm not keen on duplicate code so this is here to stay.

Reply:
On second thought, adding an extra field isn't necessary, but I would at least add parentheses to make the intention of the nested ternaries clear.